test/e2e/generator_test.go

```diff
@@ -0,0 +1,141 @@
+	t = suite.T()
+	ctx = framework.NewTestCtx(t)
+	fw = framework.Global
+	namespace, _ = ctx.GetNamespace()
+	require.NotNil(t, namespace, "GetNamespace failed")
+}
+
+func (suite *GeneratorAllInOneTestSuite) TearDownSuite() {
+	handleSuiteTearDown()
+}
+
+func TestGeneratorAllInOneSuite(t *testing.T) {
+	suite.Run(t, new(GeneratorAllInOneTestSuite))
+}
+
+func (suite *GeneratorAllInOneTestSuite) SetupTest() {
+	t = suite.T()
+}
+
+func (suite *GeneratorAllInOneTestSuite) AfterTest(suiteName, testName string) {
+	handleTestFailure()
+}
+
+// Returns deployment name and allInOne as a temporary file. Caller must delete
+func getAllInOneTempFile() (string, *os.File) {
+	cr := `apiVersion: jaegertracing.io/v1
+kind: "Jaeger"
+metadata:
+  name: "my-jaeger"
+spec:
+  strategy: allInOne
+  allInOne:
+    image: jaegertracing/all-in-one:1.13
+`
+	name := "my-jaeger"
+
+	f, err := ioutil.TempFile("", "crd*")
+	require.NoError(t, err, "temp file")
+
+	f.Write([]byte(cr))
+
+	return name, f
+}
+
+func (suite *GeneratorAllInOneTestSuite) TestAllInOne() {
+	// Get a *os.File for Jaeger CR with all in one, and the name of the deployment
+	name, cr := getAllInOneTempFile()
+	defer func() {
+		cr.Close()
+		os.Remove(cr.Name())
+	}()
+
+	// Create a temporary file for the output
+	output, err := ioutil.TempFile("", "output*")
+	require.NoError(t, err, "temp file")
+
+	defer func() {
+		output.Close()
+		os.Remove(output.Name())
+	}()
+
+	// Execute the generate command
+	generateOutput, err := exec.Command("../../build/_output/bin/jaeger-operator", "generate", "--cr", cr.Name(), "--output", output.Name()).CombinedOutput()
+	require.NoError(t, err, "generate failed: %s", generateOutput)
+
+	kubectlOutput, err := exec.Command("kubectl", "create", "-n", namespace, "-f", output.Name()).CombinedOutput()
+	require.NoError(t, err, "could not create objects from yaml: %s", kubectlOutput)
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, name, 1, retryInterval, 3*timeout)
+	require.NoError(t, err, "Error waiting for Jaeger deployment")
+
+	// Check that deployment seems OK
+	ports := []string{"0:16686", "0:14268"}
+	portForward, closeChan := CreatePortForward(namespace, name, "all-in-one", ports, fw.KubeConfig)
+	defer portForward.Close()
+	defer close(closeChan)
+	forwardedPorts, err := portForward.GetPorts()
+	require.NoError(t, err)
+
+	url := fmt.Sprintf("http://localhost:%d/search", forwardedPorts[0].Local)
+	c := http.Client{Timeout: 3 * time.Second}
+
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	require.NoError(t, err, "Failed to create httpRequest")
+
+	err = wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		res, err := c.Do(req)
+		if err != nil && strings.Contains(err.Error(), "Timeout exceeded") {
+			t.Logf("Retrying request after error %v", err)
+			return false, nil
+		}
+		require.NoError(t, err)
+
+		if res.StatusCode != 200 {
+			return false, fmt.Errorf("unexpected status code %d", res.StatusCode)
+		}
+
```

Comment:
Can you replace this with:  require.Equal(t, 200, res.StatusCode)  Using require is more concise and also gives better error messages and a nice stacktrace when the test fails.

Reply:
Thanks, fixes